source/src/main/java/com/adataptivescale/rosetta/source/core/extractors/view/KineticaViewExtractor.java

```diff
@@ -37,58 +37,62 @@
+                     "view_definition",
+                     false);
 
         //Materialized views
         processViews(connection, viewsBySchema,
-                "select * from pg_catalog.pg_matviews where schemaname='%s'",
-                "matviewname",
-                "definition",
-                true);
+                     "select * from pg_catalog.pg_matviews where schemaname='%s'",
+                     "matviewname",
+                     "definition",
+                     true);
+
+        // Skip views that were not processed, were not given any code to create
+        views.removeIf(view -> view.getCode() == null || view.getCode().isEmpty());
     }
 
     private void processViews(java.sql.Connection connection, HashMap<String, List<View>> viewsBySchema,
                               String queryTemplate, String nameField, String ddlField,
-                              boolean isMaterialized) throws SQLException {
+                              boolean isMaterialized) {
         for (String schemaName : viewsBySchema.keySet()) {
-            Statement statement = connection.createStatement();
-            String query = String.format(queryTemplate, schemaName);
-            ResultSet resultSet = statement.executeQuery(query);
-            List<Map<String, Object>> records = QueryHelper.mapRecords(resultSet);
-            for (Map<String, Object> record : records) {
-                Optional<View> tmpTable = viewsBySchema.get(schemaName).stream()
-                        .filter(view -> view.getName().equals(record.get(nameField))).findAny();
-                if (tmpTable.isPresent()) {
-                    String finalDdl = processDDL(record.get(ddlField).toString());
-                    tmpTable.get().setCode(finalDdl);
-                    if (isMaterialized) {
-                        tmpTable.get().setMaterialized(true);
+            try (Statement statement = connection.createStatement()) {
+                String query = String.format(queryTemplate, schemaName);
+                ResultSet resultSet = statement.executeQuery(query);
+                List<Map<String, Object>> records = QueryHelper.mapRecords(resultSet);
+
+                viewsBySchema.get(schemaName).forEach(view -> {
+                    Optional<Map<String, Object>> record = records.stream()
+                            .filter(rec -> view.getName().equals(rec.get(nameField)))
+                            .findAny();
+
+                    if (record.isPresent()) {
+                        String finalDdl = processDDL(record.get().get(ddlField).toString());
+                        view.setCode(finalDdl);
+                        if (isMaterialized) {
+                            view.setMaterialized(true);
+                        }
                     }
-                }
+                });
+
+            } catch (SQLException e) {
+                System.out.println("Table pg_matviews does not exist, skipping processing.");
```

Comment:
Can we make this dynamic like to use the value of **nameField** instead of static name pg_matviews. This method is called from two different places. Also we can log the SQLException e message or string. Sometimes the issue can be permissions or something else.

Reply:
Changed the message to a more dynamic error message, also modified it to log a warn message.